app/controllers/manage_evaluators_controller.rb

```diff
@@ -78,13 +78,15 @@ def fetch_existing_evaluators
 
   # Create action helpers
   def process_evaluator_invitation(email)
-    existing_invitation = @challenge.evaluator_invitations.find_by(email:, phase: @phase)
-    user = User.find_by(email:)
-
-    if existing_invitation
+    user = User.find_by(email: email)
+    existing_invitation = @challenge.evaluator_invitations.find_by(email: email, phase: @phase)
+
+    if user
+      result = add_user_as_evaluator(user)
+      existing_invitation&.destroy if result[:success]
+      result
+    elsif existing_invitation
```

Comment:
You're finding and destroying the EvaluatorInvitation record twice here, which should probably be consolidated to conditionally destroy it in `add_user_as_evaluator`

```suggestion
    user = User.find_by(email: email)
    existing_invitation = @challenge.evaluator_invitations.find_by(email: email, phase: @phase)

    if user
      add_user_as_evaluator(user, existing_invitation)
    elsif existing_invitation
```